src/client/client.ts

```diff
@@ -203,6 +326,139 @@ export default class CTProtoClient<AuthRequestPayload, AuthResponsePayload, ApiR
     });
   }
 
+  /**
+   * This method sends one chunk of the uploading file
+   *
+   * @param file - sending file
+   * @param chunkNumber - number of sending chunk
+   * @param message - additional info for chunk
+   */
+  private sendChunk(file: FileToUpload<ApiResponse['payload']>, chunkNumber: number, message: string): void {
+
+    /**
+     * Getting chunk by slicing file by the chunk number and buffer limit
+     */
+    const chunkOffset = chunkNumber * this.bufferLimit;
+    const dataOfFile = file.fileData.slice( chunkOffset, chunkOffset + this.bufferLimit );
+    const chunk = Buffer.alloc(this.bufferLimit, 0);
+
+    dataOfFile.copy(chunk, 0);
+    /**
+     * Getting info converted to binary type, which includes info about chunk number and chunk size
+     */
+    const chunkInfo = this.makeDataAboutChunk(chunkNumber, chunk.length);
+
+    /**
+     * Unite meta with file data
+     */
+    const data = Buffer.concat([chunkInfo, chunk]);
```

Comment:
And again it looks too complex:

You need to concat `fileId` + `chunkNumber` + `fileDataSize` + `fileData` + `message` as described under `file-transfer.md`

But you're doing following:

1) Concat `chunkNumber` + `fileDataSize` by the `makeDataAboutChunk()`
2) Concat the result of (1) with `fileData` in this line
3) Concat `fileId` and the result of (2) and `message` by the `MessageFactory.packChunk()`

It's hard to understand such a complex code structure. Why not just to prepare all parts first and then pack all of them via the singe method, for example `MessageFactory.packChunk`? 



It's expected to see something like this:

```js
const fileId = MessageFactory.createFileId();
const chunkNumber = ...
const fileDataSize = ...
const fileData = ...
const message = ...

const chunkReadyToUpload = MessageFactory.packChunk(fileId, chunkNumber, fileDataSize, fileData, message);

this.sendChunk(chunkReadyToUpload);

```